File: solutions/111.minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        return helper(root);
    }
    
    public int helper(TreeNode root){
        if(root == null) return 0;
        if(root.left == null && root.right == null){
            return 1;
        }else if(root.left == null){
            return helper(root.right) + 1;
        }else if(root.right == null){
            return helper(root.left) + 1;
        }
        
        return Math.min(helper(root.left), helper(root.right)) + 1;
        
    }

    
    
}

File: solutions/111.minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if(root === null) return 0;
    return BFSHelper([{
        node: root,
        deepth: 1
    }])
    
    function BFSHelper(nodeArr){
        while(nodeArr.length > 0){
            let {node, deepth} = nodeArr.shift();
            if(node.left === null && node.right === null) return deepth;
            if(node.left !== null) nodeArr.push({
                node: node.left, deepth: deepth + 1 
            });        
            if(node.right !== null) nodeArr.push({
                node: node.right,
                deepth: deepth + 1
            })
        }
        return 0;
    }

    
};